activerecord/lib/active_record/railties/databases.rake

```diff
@@ -87,9 +97,16 @@ db_namespace = namespace :db do
   desc "Migrate the database (options: VERSION=x, VERBOSE=false, SCOPE=blog)."
   task migrate: :load_config do
     original_db_config = ActiveRecord::Base.connection_db_config
-    ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env).each do |db_config|
+
+    if name = ENV["DATABASE_NAME"]
+      db_config = ActiveRecord::Base.configurations.configs_for(env_name: Rails.env, name: name)
       ActiveRecord::Base.establish_connection(db_config)
       ActiveRecord::Tasks::DatabaseTasks.migrate
```

Comment:
One way to fix the Rubocop errors regarding shadowed variables would be to use `tap`.  That could also provide nice structural parity with the `else` branch:

```suggestion
      ActiveRecord::Base.configurations.configs_for(env_name: Rails.env, name: name).tap do |db_config|
        ActiveRecord::Base.establish_connection(db_config)
        ActiveRecord::Tasks::DatabaseTasks.migrate
      end
```

Reply:
thanks @jonathanhefner!

Reply:
FYI "shadowing outer local variable" warning was removed in Ruby 2.6 so I've removed the `Lint/ShadowingOuterLocalVariable` cop in main ba97e9f918bc2ffe8ed351a0e0b8d75b67d26086.